src/quickgui/qgsquickpositionkit.h

```diff
@@ -0,0 +1,255 @@
+
+    //! \copydoc QgsQuickPositionKit::simulatePositionLongLatRad
+    QVector<double> simulatePositionLongLatRad() const;
+
+    //! \copydoc QgsQuickPositionKit::simulatePositionLongLatRad
+    void setSimulatePositionLongLatRad( const QVector<double> &simulatePositionLongLatRad );
+
+    /**
+     * Use simulated GPS source.
+     *
+     * Simulated GPS source emulates point on circle around defined point in specified radius
+     *
+     * We do not want to have the origin point as property
+     * We basically want to set it once based on project/map cente and keep
+     * it that way regardless of mapsettings change (e.g. zoom etc)
+     *
+     * \param longitude longitude of the centre of the emulated points
+     * \param latitude latitude of the centre of the emulated points
+     * \param radius distance of emulated points from the centre (in degrees WSG84)
+     */
+    Q_INVOKABLE void use_simulated_location( double longitude, double latitude, double radius );
+
+    /**
+     * Generates label for gps accuracy according set gps source. If there is no accuracy, return empty string.
+     */
+    Q_INVOKABLE QString sourceAccuracyLabel();
+
+    /**
+     * Generates label for gps position according set gps source. If has no position, return empty string.
+     * \param precision Defines number of digits after comma.
+     */
+    Q_INVOKABLE QString sourcePositionLabel( int precision );
+
+    /**
+     * Use real GPS source (not simulated)
+     */
+    Q_INVOKABLE void use_gps_location();
+
+    /**
+     * Used for changing position source when simulatePositionLongLatRad is un/set.
+     * \param simulatePositionLongLatRad Vector containing longitute, latitute and radius.
+     */
+    Q_INVOKABLE void onSimulatePositionLongLatRadChanged( QVector<double> simulatePositionLongLatRad );
+
+    /**
+     * Updates screen position according projected position.
+     */
+    Q_INVOKABLE void updateScreenPosition();
+
+  signals:
+    //! source position changed
+    void positionChanged();
+
+    //! screenPosition changed
+    void screenPositionChanged();
+
+    //! hasPosition changed
+    void hasPositionChanged();
+
+    //! changed if source position is simulated or not
+    void isSimulatedChanged();
+
+    //! \copydoc QgsQuickPositionKit::mapSettings
+    void mapSettingsChanged();
+
+    //! \copydoc QgsQuickPositionKit::simulatePositionLongLatRad
+    void simulatePositionLongLatRadChanged( QVector<double> simulatePositionLongLatRad );
+
+  public slots:
+
+  private slots:
+    void positionUpdated( const QGeoPositionInfo &info );
+    void onUpdateTimeout();
+
+  protected:
+    //! \copydoc QgsQuickPositionKit::position
+    QgsPoint mPosition;
+    //! \copydoc QgsQuickPositionKit::projectedPosition
+    QgsPoint mProjectedPosition;
+    //! \copydoc QgsQuickPositionKit::screenPosition
+    QPointF mScreenPosition;
+
+    //! \copydoc QgsQuickPositionKit::accuracy
+    qreal mAccuracy;
```

Comment:
intialize doubles/bools here

Reply:
Then there is the question, why are those not private?